core/source_map.rs

```diff
@@ -136,6 +183,25 @@ impl SourceMapper {
     }
   }
 
+  /// Load an external source map file from disk or network.
+  fn load_external_source_map_static(
+    module_url: &str,
+    source_map_url: &str,
+  ) -> Option<Vec<u8>> {
+    // Resolve source_map_url relative to module_url
+    let base_url = resolve_url(module_url).ok()?;
+    let resolved_url = base_url.join(source_map_url).ok()?;
+
+    // Only support file:// URLs for now (synchronous loading)
+    if resolved_url.scheme() == "file" {
+      let path = resolved_url.to_file_path().ok()?;
+      std::fs::read(path).ok()
```

Comment:
We can't do that like this - if you need to read something from disk this will need to be delegated to `ModuleLoader` to ensure permission checks are performed - I think it might be fine to leave it for a follow up PR for now.

Reply:
So this should essentially do `self.module_loader.load_external_source_map()`